File: src/py_pkg/py_pkg/simple_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node


class SimpleNode(Node):
    def __init__(self):
        super().__init__('simple_node_py')
        self.get_logger().info("Simple Python Node has been created")
        hz = 0.2 #period = 1/0.2 = 5 seconds
        timer_ = self.create_timer(1/hz, self.timer_callback) 

    def timer_callback(self):
        self.get_logger().info("Hello from node")

def main(args=None):
    rclpy.init(args=args)
    node = SimpleNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__=='__main__':
    main()



    

File: src/cpp_py_pkg/src/simple_node.cpp
#include "rclcpp/rclcpp.hpp"
#include <chrono>

class SimpleNode : public rclcpp::Node
{
public:
  SimpleNode(): Node("simple_node_cpp")
  {
    RCLCPP_INFO(this->get_logger(), "Simple C++ Node from cpp_py_pkg has been created");
    timer_ = this->create_wall_timer(std::chrono::milliseconds(500),std::bind(&SimpleNode::timer_callback, this)
);
  }
  private:
    void timer_callback()
    {
      RCLCPP_INFO(this->get_logger(), "Hello from node");
    }
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}